core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -605,4 +605,26 @@ private static void setFileAttributes(final Path path, final Set<PosixFilePermis
             Files.setPosixFilePermissions(path, permissions);
         }
     }
+
+    private final List<Path> pathsToDeleteOnShutdown = new ArrayList<>();
+
+    @Override
+    public void close() throws IOException {
+        IOException exception = null;
+        for (final Path pathToDeleteOnShutdown : pathsToDeleteOnShutdown) {
+            try {
+                IOUtils.rm(pathToDeleteOnShutdown);
```

Comment:
`rm` takes multiple paths, why not use that? It handles exceptions from any of them.